### ISSUE TITLE
unit groups and unit settings should be restored after loading a game

### ISSUE THREAD

**Issue by deee167** [ZeroK-RTS/Zero-K#4175]
unit groups and unit settings should be restored after loading a game.  Right now unit groups for sure aren't being loaded from a save.

**Comment by GoogleFrog** [comment]
As far as I know lua can't do anything about this.

**Comment by deee167** [comment]
> 
> 
> As far as I know lua can't do anything about this.

There is no way to do this?  Are you sure this was never working before?

**Comment by GoogleFrog** [comment]
That isn't what I said. Are you going to look into it?

**Comment by deee167** [comment]
> 
> 
> That isn't what I said. Are you going to look into it?

What are you talking about?  I'm not a Zero-K developer.  I'm a player.  I don't code Zero-K.  I'm reporting a bug to make the game better.

Thanks

**Comment by Arch-Shaman** [comment]
Everyone can look into bugs, even nondevelopers. While you may not code ZK, you can contribute by investigating bugs or looking into solutions.

> As far as I know lua can't do anything about this.

Spring.SetUnitGroup (needs ModUICtrl)

 ( number unitID, number groupID ) -> nil

Combine this with
Spring.GetGroupUnits() on save.

**Comment by deee167** [comment]
> 
> 
> Everyone can look into bugs, even nondevelopers. While you may not code ZK, you can contribute by investigating bugs or looking into solutions.
> 
> > As far as I know lua can't do anything about this.
> 
> Spring.SetUnitGroup (needs ModUICtrl)
> 
> ( number unitID, number groupID ) -> nil
> 
> Combine this with
> Spring.GetGroupUnits() on save.

I am also contributing by reporting bugs and providing feedback.  I am also discussing the issues with you guys and investigating the behaviours.

**Comment by Arch-Shaman** [comment]
Yes, if it works it gets stabled and everyone gets it. I'll add this to commshare too, since IIRC commsharing players don't carry over their unitgroups.

**Comment by deee167** [comment]
I noticed another issue with save and load.  Not sure if you want another ticket for it or not.  The issue is some or all buildings don't have their collision loaded properly after a save and load.  For eg. Building some Lucifer turrets, saving the game, then loading the game results in the following issue:
When you are in build preview and trying to build more buildings, eg. more Lucifers, the game shows a green grid indicating that you can build the new building even when you are right overtop of the previously build Lucifers that you just loaded.

**Comment by Arch-Shaman** [comment]
Should open another ticket for that. Probably would be nice if @sprunk could kindly add a save/load tag too?

**Comment by deee167** [comment]
Any luck?

**Comment by deee167** [comment]
Did you make a change to the game?  It might not be related, but I can't load any of my saves anymore, it crashes the game.

**Comment by deee167** [comment]
save load is broken any update?

**Comment by segatrade** [comment]
> Did you make a change to the game? It might not be related, but I can't load any of my saves anymore, it crashes the game.

You need use same game version for save and load. in steam launch options example `zkmenu:stable 2025.06.20`
https://zero-k.info/Forum/Thread/36102 more

**Comment by segatrade** [comment]
> unit groups and unit settings should be restored after loading a game. Right now unit groups for sure aren't being loaded from a save.

This is still bug in zero-k on 2025.06.20 engine. It's issue for this repo or where better?
